stacks/MyStack.ts

```diff
@@ -10,6 +16,68 @@ export function MyStack({ stack }: StackContext) {
   const { eventBus } = use(EventBusStack);
   const functionProps = getFunctionProps();
 
+  const schedulerRole = new Role(stack, "LambdaVPCRole", {
+    assumedBy: new ServicePrincipal("lambda.amazonaws.com"),
+  });
+
+  schedulerRole.addToPolicy(
+    new PolicyStatement({
+      actions: [
+        "scheduler:CreateSchedule",
+        "scheduler:GetSchedule",
+        "scheduler:UpdateSchedule",
+        "scheduler:DeleteSchedule",
+        "iam:PassRole",
+      ],
+      resources: ["*"],
+    }),
+  );
+
+  const scheduleHandlerLambda = new Function(stack, "ScheduleHandlerLambda", {
+    handler: "packages/functions/schedule/scheduleHandlerLambda.handler",
+    bind: [...functionProps.bind, eventBus],
+    role: new Role(stack, "PutEventsRole", {
+      assumedBy: new ServicePrincipal("lambda.amazonaws.com"),
+      inlinePolicies: {
+        SchedulerPolicy: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              actions: ["events:PutEvents"],
+              resources: ["*"],
+            }),
+          ],
+        }),
+      },
+    }),
+    permissions: [eventBus],
+    environment: {
+      EVENT_BUS_NAME: eventBus.eventBusName,
+    },
+  });
+  scheduleHandlerLambda.addPermission("AllowEventBridgeInvoke", {
+    principal: new ServicePrincipal("events.amazonaws.com"),
+    action: "lambda:InvokeFunction",
+  });
+
+  const schedulerFunctionRole = new Role(
+    stack,
+    `schedulerFunctionRole-${stack.stage}`,
+    {
+      assumedBy: new ServicePrincipal("scheduler.amazonaws.com"),
+      inlinePolicies: {
+        lambdaExecute: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              actions: ["lambda:InvokeFunction"],
+              resources: ["*"],
+            }),
+          ],
+        }),
+      },
+      roleName: `schedulerFunctionRole-${stack.stage}`,
+    },
+  );
+
```

Comment:
Do we really need all of these? 😨 I thought adding eventBus to permissions would be enough.